01-JS-DOM-APIs/js/javascript.js

```diff
@@ -0,0 +1,193 @@
+
+function appearInFullLoad(){ /* Show the 'hello world' when the page is fully loaded */
+
+setTimeout(function() {
+  
+    const element = document.getElementById("hide");
+    element.classList.remove("hidden");              
+    },700);
+}
+
+
+function getJoke(){
+
+fetch("http://api.icndb.com/jokes/random")
+
+.then(function(response) { 
+return response.json();
+})
+
+.then(function(myJson) {            
+
+showTheJoke(myJson);
+
+})
+
+.catch(function(error){
+
+errorDetected();
+
+})
+}
+
+
+//the function to show the joke in the html
+
+function showTheJoke(myJson){
+document.getElementById("jokeHere").innerHTML= myJson.value.joke;
+}
+
+
+//the function changes the background to red in the section of the joke
+function errorDetected(){
+
+document.getElementById("jokeZone").style.backgroundColor="red";
+
+/* you can also use the function below if you also want to show the user that an error happened in the section of the joke
+document.getElementById("jokeHere").innerHTML= "An error has occurred";
+*/
+
+}
+
+
+// point 4
+
+
+function search(){
```

Comment:
An improvement to this kind of calls, since you make two of them, is to make a general function to make the API calls. so you avoid repeating code to make these calls. and only pass relevant information such as URI, query parameters, etc.. and to the result of that function attached the then, catch for the particular case.